fs/copy_test.ts

```diff
@@ -0,0 +1,560 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+import { test } from "../testing/mod.ts";
+import {
+  assertEquals,
+  assertThrows,
+  assertThrowsAsync,
+  assert
+} from "../testing/asserts.ts";
+import { copy, copySync } from "./copy.ts";
+import { exists, existsSync } from "./exists.ts";
+import * as path from "./path/mod.ts";
+import { ensureDir, ensureDirSync } from "./ensure_dir.ts";
+import { ensureFile, ensureFileSync } from "./ensure_file.ts";
+import { ensureSymlink, ensureSymlinkSync } from "./ensure_symlink.ts";
+
+const testdataDir = path.resolve("fs", "testdata");
+
+// TODO(axetroy): Add test for Windows once symlink is implemented for Windows.
+const isWindows = Deno.platform.os === "win";
+
+test({
+  name: "[fs] copy file if it does no exist",
+  async fn(): Promise<void> {
+    const srcFile = path.join(testdataDir, "copy_file_not_exists.txt");
+    const destFile = path.join(testdataDir, "copy_file_not_exists_1.txt");
+    await assertThrowsAsync(
+      async (): Promise<void> => {
+        await copy(srcFile, destFile);
```

Comment:
You're making changes to the source tree. This shouldn't happen. Make the assumption that the source tree is immutable, and create a tempdir to perform your tests.

(I realize that some of the other tests do this too. They need to be fixed.)

Reply:
yes.

Now in every test case, it should create a tempdir to store the output file.